### ISSUE TITLE
Cannot visualize lcov coverage data

### ISSUE THREAD

**Issue by abarth** [flutter/flutter-intellij#718]
The `flutter test --coverage` command produces an `coverage/lcov.info` file that contains information about the line coverage of your tests.  We should be able to visualize that information in IntelliJ.

1. When editing a file, you should be able to see whether a line was (a) executed in a test, (b) possible to execute but not actually executed, (c) not actually possible to execute.  (All this data is in the lcov file.)

2. There should be some sort of overview panel that lists the various files in the project and what percent line coverage they have.  Also, there should be an overall readout for the percent coverage of the whole project.

For example, the [lcov-info](https://atom.io/packages/lcov-info) plugin does this for Atom.  It's likely that such a plugin already exists for IntelliJ.  We might simply need to find it and recommend.

**Comment by eseidel** [comment]
See https://github.com/angular/angular-cli/issues/3560#issuecomment-270141272 for possible inspiration.

**Comment by zoechi** [comment]
IntelliJ already supports "Run with coverage" for Dart 
and renders executed code with different background colors 
and also shows coverage percentage besides each file in the Project window.
Would be great to have that in Flutter.

**Comment by MisterJimson** [comment]
I was able to visualize with this in VS Code https://github.com/ryanluker/vscode-coverage-gutters

**Comment by Lootwig** [comment]
It's a shame this feature is not available for flutter, in spite of plain dart being supported for a long time now.

**Comment by tvolkert** [comment]
/cc @devoncarew 

Running into this while trying to visualize coverage in a pending PR.

**Comment by md-weber** [comment]
There is a coverage option in Android Studio and IntelliJ that I usually use that will add for each line the information of the lcov file. That works amazingly good in react and other JS libraries.

<img width="1176" alt="grafik" src="https://user-images.githubusercontent.com/8026644/99043229-54030580-258e-11eb-83dd-436b59db79b4.png">

**Comment by yulkin2002** [comment]
is there any update on on a coverage visualization plugin for Flutter?

**Comment by maximilianmaihoefner** [comment]
I just found https://github.com/marius-h/flutter_enhancement_suite which has support for running flutter tests with coverage and visualizes the lcov.info. (Only the 1.4.0 Version is in the Jetbrains Plugin Repo, but the Feature was added with 1.5.0 which was released just a few hours ago, so until the update is approved by jetbrains you have to install the plugin manually from the github release page)

**Comment by marius-h** [comment]
@maximilianmaihoefner I'm pretty sure the update will be available on Monday :) Let me know if the feature works for you or if you encountered any issues

**Comment by md-weber** [comment]
This issue can be closed as soon as Jetbrains approves the update. The solution for the problem is to use the [Flutter Enhancement Suite](https://github.com/marius-h/flutter_enhancement_suite) of @marius-h. 

## Install before the update is approved:
1. Download the Plugin 1.5.0 in the [release section](https://github.com/marius-h/flutter_enhancement_suite/releases/tag/v1.5.0)
2. In Android Studio / IntelliJ navigate to the Plugins section
3. Select the gear icon and choose the Install plugin from the disk
![grafik](https://user-images.githubusercontent.com/8026644/114268484-9e839380-9a01-11eb-9a99-fbe745f3373b.png)
4. Select the zip file

## Install after the update
You can find the plugin in Android Studio Plugins -> Marketplace -> Search for "Flutter Enhancement Suite" -> Download and Restart. Or via the [Plugin Page](https://plugins.jetbrains.com/plugin/12693-flutter-enhancement-suite). 

![grafik](https://user-images.githubusercontent.com/8026644/114268532-d38fe600-9a01-11eb-8031-623cff55a9e2.png)

**Comment by stevemessick** [comment]
Part 1 is fixed in #5463. Not sure when we'll get to part 2, so keeping this open.

**Comment by GroovinChip** [comment]
Any progress on this? Visualizing the coverage would be super helpful.

**Comment by stevemessick** [comment]
@GroovinChip can you be more specific?

**Comment by GroovinChip** [comment]
@stevemessick yes, being able to view the coverage in the line gutters would be quite helpful. Currently I have to manually generate HTML out of my lcov.info and view it in the browser to view highlights. This highlights whole lines, though, not the gutters. Having red/green lines in the line gutters would be really helpful.

Additionally, having a percentage of lines covered showing on filenames would be very helpful.

**Comment by stevemessick** [comment]
Red/green highlighting is supported as of #5463. You need to run the tests from IntelliJ using the "run with coverage" button, but after that everything should be automatically displayed. Make sure you have the latest plugin and give that a try. If you run into problems, I suggest opening a new issue, as this one is now just tracking the second request.

And, your second paragraph echoes the second request. We're working on a table view of percentages. It got side-lined due to re-prioritization but will eventually be implemented.

**Comment by GroovinChip** [comment]
> You need to run the tests from IntelliJ using the "run with coverage" button, but after that everything should be automatically displayed

I have been doing this

> Make sure you have the latest plugin and give that a try

I do have the latest plugin, the one released today

**Comment by cedric-booxi** [comment]
[Flutter Enhancement Suite](https://github.com/marius-h/flutter_enhancement_suite) dropped coverage support.

**Comment by gentunian** [comment]
it's frustrating how as a paid user of intellij idea ultimate the feature for reading a simple file like `lcov.info` and highlighting lines is not available as default. Really, the `lcov.info` file specifically tell you which file, which line, and everything:
```

SF:lib/src/cfoo.dart
DA:25,2
DA:35,1
DA:38,3
DA:39,1
DA:40,1
DA:41,1
DA:44,2
DA:45,2
DA:46,1
DA:50,1
DA:52,2
DA:56,2
DA:57,2
DA:60,1
DA:62,4
LF:15
LH:15
end_of_record
```

how is it there's no support for this?
